app/screens/browse_channels/browse_channels.tsx

```diff
@@ -0,0 +1,501 @@
+    const ids = joinedChannels.map((c) => c.id);
+    return allChannels?.filter((c) => !ids.includes(c.id));
+};
+
+const makeLeftButton = (icon: ImageResource): OptionsTopBarButton => {
+    return {
+        id: 'close-more-channels',
+        icon,
+        testID: 'close.more_channels.button',
+    };
+};
+
+const makeRightButton = (theme: Theme, formatMessage: IntlShape['formatMessage']): OptionsTopBarButton => {
+    return {
+        color: theme.sidebarHeaderTextColor,
+        id: 'create-pub-channel',
+        text: formatMessage({id: 'mobile.create_channel', defaultMessage: 'Create'}),
+        showAsAction: 'always',
+        testID: 'more_channels.create.button',
+    };
+};
+
+const close = () => {
+    Keyboard.dismiss();
+    dismissModal();
+};
+
+const channelKeyExtractor = (channel: Channel) => {
+    return channel.id;
+};
+const getStyleFromTheme = makeStyleSheetFromTheme((theme: Theme) => {
+    return {
+        container: {
+            flex: 1,
+        },
+        searchBar: {
+            marginHorizontal: 12,
+            borderRadius: 8,
+            marginTop: 12,
+            backgroundColor: changeOpacity(theme.centerChannelColor, 0.08),
+        },
+        noResultContainer: {
+            flexGrow: 1,
+            alignItems: 'center',
+            justifyContent: 'center',
+        },
+        noResultText: {
+            fontSize: 26,
+            color: changeOpacity(theme.centerChannelColor, 0.5),
+        },
+        searchBarInput: {
+            color: theme.centerChannelColor,
+        },
+        loadingContainer: {
+            flex: 1,
+            justifyContent: 'center',
+            alignItems: 'center',
+        },
+        loading: {
+            height: 32,
+            width: 32,
+            justifyContent: 'center',
+        },
+        listContainer: {
+            paddingHorizontal: 20,
+            flexGrow: 1,
+        },
+    };
+});
+
+export default function BrowseChannels({
+    componentId,
+    canCreateChannels = false,
+    joinedChannels = [],
+    sharedChannelsEnabled = false,
+    closeButton,
+    currentUserId,
+    currentTeamId,
+    canShowArchivedChannels = false,
+    categoryId,
+}: Props) {
+    const intl = useIntl();
+    const theme = useTheme();
+    const style = getStyleFromTheme(theme);
+    const serverUrl = useServerUrl();
+
+    const [channels, setChannels] = useState<Channel[]>([]);
+    const [archivedChannels, setArchivedChannels] = useState<Channel[]>([]);
+    const [sharedChannels, setSharedChannels] = useState<Channel[]>([]);
+
+    const [visibleChannels, setVisibleChannels] = useState<Channel[]>([]);
+    const [typeOfChannels, setTypeOfChannels] = useState(PUBLIC);
+
+    const [term, setTerm] = useState('');
+
+    const [adding, setAdding] = useState(false);
+    const [loading, setLoading] = useState(false);
+
+    const mounted = useRef(false);
```

Comment:
highly doubt this is needed.

Reply:
My fear (and why I use it) is if we close the screen while it is loading channels. It may call the some setX, which would be a state change on an unmounted component. That might be handled beautifully by react, but wanted to fall on the "safe side".